cli/compilers/json.rs

```diff
@@ -1,32 +1,30 @@
 // Copyright 2018-2020 the Deno authors. All rights reserved. MIT license.
 use crate::compilers::CompiledModule;
-use crate::compilers::CompiledModuleFuture;
 use crate::file_fetcher::SourceFile;
-use crate::futures::future::FutureExt;
 use deno_core::ErrBox;
 use regex::Regex;
-use std::pin::Pin;
-use std::str;
 
 // From https://github.com/mathiasbynens/mothereff.in/blob/master/js-variables/eff.js
 static JS_RESERVED_WORDS: &str = r"^(?:do|if|in|for|let|new|try|var|case|else|enum|eval|false|null|this|true|void|with|await|break|catch|class|const|super|throw|while|yield|delete|export|import|public|return|static|switch|typeof|default|extends|finally|package|private|continue|debugger|function|arguments|interface|protected|implements|instanceof)$";
 
 pub struct JsonCompiler {}
 
 impl JsonCompiler {
-  pub fn compile_async(
+  pub async fn compile_async(
     &self,
     source_file: &SourceFile,
-  ) -> Pin<Box<CompiledModuleFuture>> {
+  ) -> Result<CompiledModule, ErrBox> {
     let maybe_json_value: serde_json::Result<serde_json::Value> =
-      serde_json::from_str(&str::from_utf8(&source_file.source_code).unwrap());
+      serde_json::from_str(
```

Comment:
While you're at it, maybe remove the str::from_utf8 part and use serde_json::from_slice ?

Reply:
done